### DISCUSSION TITLE
error C2065: 'anet_type': undeclared identifier , any help on this

### DISCUSSION THREAD

**Discussion by ranjith502** [davisking/dlib#2787] [Q&A]
iam trying emotion detection encountering below error any help

#include <dlib/dnn.h>
#include <dlib/gui_widgets.h>
#include <dlib/clustering.h>
#include <dlib/string.h>
#include <dlib/image_io.h>
#include <dlib/image_processing/frontal_face_detector.h>

using namespace dlib;
using namespace std;

int main()
{
    try
    {
        // Load face detection model
        frontal_face_detector detector = get_frontal_face_detector();
        // Load shape predictor model
        shape_predictor sp;
        deserialize("shape_predictor_68_face_landmarks.dat") >> sp;

        // Load emotion recognition model
        anet_type net;
        deserialize("dlib_face_recognition_resnet_model_v1.dat") >> net;

        // Load the input image
        array2d<unsigned char> img;
        load_image(img, "test.jpg");

        // Detect faces in the image
        std::vector<rectangle> dets = detector(img);
        cout << "Number of faces detected: " << dets.size() << endl;

        // Detect landmarks on each face
        std::vector<full_object_detection> shapes;
        for (unsigned long j = 0; j < dets.size(); ++j)
        {
            full_object_detection shape = sp(img, dets[j]);
            shapes.push_back(shape);
        }

        // Compute emotion for each face
        for (unsigned long i = 0; i < shapes.size(); ++i)
        {
            matrix<rgb_pixel> face_chip;
            extract_image_chip(img, get_face_chip_details(shapes[i], 150, 0.25), face_chip);
            matrix<float, 0, 1> face_descriptor = net(face_chip);
            cout << "Emotions for face " << i+1 << ":" << endl;
            cout << "Angry: " << face_descriptor(0) << endl;
            cout << "Happy: " << face_descriptor(1) << endl;
            cout << "Neutral: " << face_descriptor(2) << endl;
            cout << "Sad: " << face_descriptor(3) << endl;
        }
    }
    catch(exception& e)
    {
        cout << e.what() << endl;
    }
    return 0;
}

**Comment by arrufat** [comment]
That `anet_type` is a custom user-defined type for a network, you need to define it yourself in your code, check this example: http://dlib.net/dnn_face_recognition_ex.cpp.html

**Reply by ranjith502** [reply]
@arrufat thanks brother

**Reply by LucianoRoda** [reply]
Hi, i've got the same problem and i coudn't fix it, @ranjith502 did you fix it?
